zmq/backend/cython/socket.pyx

```diff
@@ -633,44 +633,53 @@ cdef class Socket:
         """
         cdef int rc
         cdef char* c_addr
-        
+
         _check_version((3,2), "disconnect")
         _check_closed(self)
         if isinstance(addr, unicode):
             addr = addr.encode('utf-8')
         if not isinstance(addr, bytes):
             raise TypeError('expected str, got: %r' % addr)
         c_addr = addr
-        
+
         rc = zmq_disconnect(self.handle, c_addr)
         if rc != 0:
             raise ZMQError()
 
-    def monitor(self, addr, int events=ZMQ_EVENT_ALL):
+    def monitor(self, addr, int events=-1):
         """s.monitor(addr, flags)
 
         Start publishing socket events on inproc.
         See libzmq docs for zmq_monitor for details.
-        
+
         While this function is available from libzmq 3.2,
         pyzmq cannot parse monitor messages from libzmq prior to 4.0.
-        
+
         .. versionadded: libzmq-3.2
         .. versionadded: 14.0
-        
+
+        .. versionchanged: 18.0
+            default to new zmq.PYZMQ_EVENT_ALL,
+            which is all events known by pyzmq,
+            instead of zmq.EVENT_ALL,
```

Comment:
note that we are changing this, and making the EVENT_ALL fixed - the monitor stuff is now versioned independently, and if you don't ask for the new version you won't get new events in the future

Reply:
Thanks. I went ahead and reverted this, leaving it up to libzmq for how to behave.